lib/model/content.dart

```diff
@@ -504,6 +504,84 @@ class EmbedVideoNode extends BlockContentNode {
   }
 }
 
+class TableNode extends BlockContentNode {
+  const TableNode({super.debugHtmlNode, required this.rows});
+
+  final List<TableRowNode> rows;
+
+  @override
+  List<DiagnosticsNode> debugDescribeChildren() {
+    return rows
+      .mapIndexed((i, row) => row.toDiagnosticsNode(name: 'row $i'))
+      .toList();
+  }
+}
+
+class TableRowNode extends BlockContentNode {
+  const TableRowNode({
+    super.debugHtmlNode,
+    required this.cells,
+    required this.isHeader,
+  });
+
+  final List<TableCellNode> cells;
+
+  /// Indicates whether this row is the header row.
+  final bool isHeader;
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties.add(FlagProperty('isHeader', value: isHeader, ifTrue: "is header"));
+  }
+
+  @override
+  List<DiagnosticsNode> debugDescribeChildren() {
+    return cells
+      .mapIndexed((i, cell) => cell.toDiagnosticsNode(name: 'cell $i'))
+      .toList();
+  }
+}
+
+// The text-alignment setting that applies to a cell's column, from the delimiter row.
+//
+// See GitHub-flavored Markdown:
+//   https://github.github.com/gfm/#tables-extension-
+enum TableColumnTextAlignment {
+  /// All cells' text left-aligned, represented in Markdown as `|: --- |`.
+  left, // TODO(i18n) RTL issues? https://github.com/zulip/zulip/issues/32265
+  /// All cells' text center-aligned, represented in Markdown as `|: --- :|`.
+  center,
+  /// All cells' text right-aligned, represented in Markdown as `| --- :|`.
+  right, // TODO(i18n) RTL issues? https://github.com/zulip/zulip/issues/32265
+  /// Cells' text aligned the default way, represented in Markdown as `| --- |`.
+  defaults
+}
+
+class TableCellNode extends BlockInlineContainerNode {
+  const TableCellNode({
+    super.debugHtmlNode,
+    required super.nodes,
+    required super.links,
+    required this.textAlignment,
+  });
+
+  /// The table column text-alignment to be used for this cell.
+  // In Markdown, alignment is defined per column using the delimiter row.
+  // However, the generated HTML specifies alignment for each cell in a row
+  // individually, that matches the UI widget implementation which is also
+  // row based and needs alignment information to be per cell.
+  final TableColumnTextAlignment textAlignment;
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties.add(EnumProperty<TableColumnTextAlignment>(
+      'textAlignment', textAlignment,
```

Comment:
nit: type can be inferred
```suggestion
    properties.add(EnumProperty('textAlignment', textAlignment,
```